lib/js/src/manager/screen/utils/_PresentAlertOperation.js

```diff
@@ -0,0 +1,501 @@
+            } else {
+                console.log('All alert audio files uploaded');
+            }
+            return true;
+        }
+    }
+
+    /**
+     * Checks whether the _AlertView has images that need to be uploaded, then uploads them.
+     * @returns {Promise} - resoves to Boolean.
+     */
+    async uploadImages () {
+        const artworksToBeUploaded = [];
+
+        if (this.supportsAlertIcon() && this._fileManager !== null && this._fileManager.fileNeedsUpload(this._alertView.getIcon())) {
+            artworksToBeUploaded.push(this._alertView.getIcon());
+        }
+
+        if (this._alertView.getSoftButtons() !== null && this._alertView.getSoftButtons() !== undefined) {
+            for (const object of this._alertView.getSoftButtons()) {
+                if (this.supportsSoftButtonImages() && object.getCurrentState() !== null && object.getCurrentState() !== undefined && this._fileManager !== null && this._fileManager.fileNeedsUpload(object.getCurrentState().getArtwork())) {
+                    artworksToBeUploaded.push(object.getCurrentState().getArtwork());
+                }
+            }
+        }
+
+        return await this.sendImages(artworksToBeUploaded);
+    }
+
+    /**
+     * Uploads an array of images to the module.
+     * @param {SdlFile[]} images - An array of images to be uploaded.
+     * @returns {Promise} - Whether the images were uploaded successfully.
+     */
+    async sendImages (images) {
+        if (images.length === 0) {
+            console.log('No Images to upload for alert');
+            return true;
+        }
+
+        console.log('Uploading images for alert');
+
+        if (this._fileManager !== null && this._fileManager !== undefined) {
+            const successes = await this._fileManager.uploadArtworks(images);
+            if (this.getState() === _Task.CANCELED) {
+                console.log('Operation canceled');
+                this._finishOperation(false, null);
+                return false;
+            }
+
+            if (successes !== null && successes !== undefined && successes.includes(false)) {
+                console.log('AlertManager artwork failed to upload with error');
+                return false;
+            } else {
+                console.log('All alert images uploaded');
+                return true;
+            }
+        }
+    }
+
+    /**
+     * Presents the Alert in the window.
+     */
+    async presentAlert () {
+        const alert = this.alertRpc();
+        const response = await this._lifecycleManager.sendRpcResolve(alert);
+        if (!response.getSuccess()) {
+            console.log(`There was an error presenting the alert: ${response.getInfo()}`);
+        } else {
+            console.log('Alert finished presenting');
+        }
+        this._finishOperation(response.getSuccess(), response.getTryAgainTime());
+    }
+
+    /**
+     * Sends a CancelInteraction to cancel the Alert.
+     */
+    async cancelAlert () {
```

Comment:
We got rid of `cancelInteraction` method and added the logic here, We also changed a lot of logic and utilized the new `isAlertPresented` flag